planner/core/task.go

```diff
@@ -209,6 +211,21 @@ func finishCopTask(ctx sessionctx.Context, task task) task {
 	newTask := &rootTask{
 		cst: t.cst,
 	}
+	if t.idxMergePartPlans != nil {
+		p := PhysicalIndexMergeReader{partialPlans: t.idxMergePartPlans, tablePlan: t.tablePlan}.Init(ctx)
+		if t.tablePlan != nil {
+			p.stats = t.tablePlan.statsInfo()
+		} else {
+			var totalRowCount float64
+			for _, partPlan := range t.idxMergePartPlans {
+				totalRowCount += partPlan.StatsCount()
+			}
+			p.stats = property.NewSimpleStats(totalRowCount)
+			p.stats.StatsVersion = t.idxMergePartPlans[0].statsInfo().StatsVersion
```

Comment:
How about using `t.idxMergePartPlans[0].statsInfo().Scale()` to keep the NDV of the stats info? Also, we can put the stats assignment into `Init` as well?

Reply:
Can we put the stats assignment into `Init` as well?